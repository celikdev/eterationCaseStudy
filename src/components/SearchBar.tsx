import {View, Text, TextInput} from 'react-native';
import React from 'react';
import {SearchIcon} from '../assets/icons/Icons';
import {COLORS} from '../constants/Colors';
import {useDispatch, useSelector} from 'react-redux';
import {setItemList, setSearchText} from '../redux/reducer/itemListReducer';

const SearchBar = () => {
  const dispatch = useDispatch();
  const searchText = useSelector((state: any) => state.itemList.searchText);

  return (
    <View
      style={{
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        borderWidth: 2,
        borderColor: COLORS.gray,
        marginHorizontal: '4%',
        borderRadius: 10,
        marginVertical: '2%',
        height: 50,
      }}>
      <SearchIcon fill={undefined} width={undefined} height={undefined} />
      <TextInput
        onChangeText={text => dispatch(setSearchText(text))}
        placeholder="Search"
        placeholderTextColor={COLORS.darkGray}
        style={{
          fontFamily: 'Montserrat-SemiBold',
          fontSize: 14,
          width: '86%',
          color: 'black',
          fontWeight: 'bold',
        }}
      />
    </View>
  );
};

export default SearchBar;
