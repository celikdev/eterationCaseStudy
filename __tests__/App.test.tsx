import React from 'react';
import {describe, it, expect} from '@jest/globals';
import {render, fireEvent} from '@testing-library/react-native';
import SearchBar from '../src/components/SearchBar';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('SearchBar component', () => {
  it('dispatches setSearchText correctly when TextInput value changes', () => {
    const store = mockStore({
      itemList: {
        searchText: '',
      },
    });

    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <SearchBar />
      </Provider>,
    );

    const inputElement = getByPlaceholderText('Search');

    fireEvent.changeText(inputElement, 'New Search Text');

    const actions = store.getActions();
    expect(actions).toEqual([
      {type: 'itemList/setSearchText', payload: 'New Search Text'},
    ]);
  });
});
