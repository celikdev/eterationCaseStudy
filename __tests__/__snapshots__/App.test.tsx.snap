// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="button"
>
  <Text>
    Press me
  </Text>
</View>
`;
